builders/gradle.go

```diff
@@ -0,0 +1,149 @@
+package builders
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	logging "github.com/op/go-logging"
+
+	"github.com/fossas/fossa-cli/config"
+	"github.com/fossas/fossa-cli/module"
+)
+
+var gradleLogger = logging.MustGetLogger("gradle")
+
+// GradleBuilder implements Builder for build.gradle builds
+type GradleBuilder struct {
+	GradleCmd     string
+	GradleVersion string
+}
+
+// Initialize collects metadata on Java and Maven binaries
+func (builder *GradleBuilder) Initialize() error {
+	gradleLogger.Debug("Initializing Gradle builder...")
+
+	// Set Gradle context variables
+	gradleCmd, gradleVersionOut, err := which("--version --offline", os.Getenv("GRADLE_BINARY"), "./gradlew", "gradle")
+	if err == nil {
+		builder.GradleCmd = gradleCmd
+
+		gradleVersionMatchRe := regexp.MustCompile(`Gradle ([0-9]+\.[0-9]+.\w+)`)
```

Comment:
It's probably more useful (for debugging purposes) to just capture the entire `--version` output rather than checking for a version.